src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/NoWhitespaceAfterCheck.java

```diff
@@ -67,6 +70,12 @@
  *     &lt;property name="allowLineBreaks" value="false"/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>
+ * If the annotation is between the type and the array, the check will skip validation for spaces:
+ * </p>
+ * <pre>
+ * public void a(final char @NotNull [] param) {} // No violation
```

Comment:
"a" ==> "foo".
Do not show bad practices to user, it is even hard to read.

Reply:
@romani , fixed.